x-pack/plugins/apm/server/routes/settings/agent_configuration.ts

```diff
@@ -102,55 +103,47 @@ export const agentConfigurationAgentNameRoute = createRoute(() => ({
     const setup = await setupRequest(context, request);
     const { serviceName } = context.params.query;
     const agentName = await getAgentNameByService({ serviceName, setup });
-    return agentName;
+    return { agentName };
   }
 }));
 
-export const createAgentConfigurationRoute = createRoute(() => ({
-  method: 'POST',
-  path: '/api/apm/settings/agent-configuration/new',
-  params: {
-    body: agentPayloadRt
-  },
+export const createOrUpdateAgentConfigurationRoute = createRoute(() => ({
+  method: 'PUT',
+  path: '/api/apm/settings/agent-configuration',
   options: {
     tags: ['access:apm', 'access:apm_write']
   },
+  params: {
+    query: t.partial({ overwrite: jsonRt.pipe(t.boolean) }),
```

Comment:
should we create an issue for handling this in the middleware layer?

Reply:
Yes, good idea

Reply:
https://github.com/elastic/kibana/issues/58411